pkg/applyinator/applyinator.go

```diff
@@ -97,106 +118,250 @@ func checksum(input []byte) string {
+	if runOneTimeInstructions {
+		executionOutputs := map[string][]byte{}
+		if len(existingPeriodicOutput) > 0 {
+			buffer := bytes.NewBuffer(existingOneTimeOutput)
+			gzReader, err := gzip.NewReader(buffer)
+			if err != nil {
+				return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+			}
+
+			var objectBuffer bytes.Buffer
+			_, err = io.Copy(&objectBuffer, gzReader)
+			if err != nil {
+				return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+			}
+
+			if err := json.Unmarshal(objectBuffer.Bytes(), &executionOutputs); err != nil {
+				return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+			}
+		}
+
+		for index, instruction := range cp.Plan.OneTimeInstructions {
+			logrus.Debugf("[Applyinator] Executing instruction %d for plan %s", index, cp.Checksum)
+			executionInstructionDir := filepath.Join(executionDir, cp.Checksum+"_"+strconv.Itoa(index))
+			prefix := cp.Checksum + "_" + strconv.Itoa(index)
+			output, _, _, err := a.execute(ctx, prefix, executionInstructionDir, instruction.CommonInstruction, true)
+			if err != nil {
+				logrus.Errorf("error executing instruction %d: %v", index, err)
+				oneTimeApplySucceeded = false
+			}
+			if instruction.Name == "" && instruction.SaveOutput {
+				logrus.Errorf("instruction does not have a name set, cannot save output data")
+			} else if instruction.SaveOutput {
+				executionOutputs[instruction.Name] = output
+			}
+			// If we have failed to apply our one-time instructions, we need to break in order to stop subsequent instructions from executing.
+			if !oneTimeApplySucceeded {
+				break
+			}
+		}
+		var gzOutput bytes.Buffer
+
+		gzWriter := gzip.NewWriter(&gzOutput)
+
+		marshalledExecutionOutputs, err := json.Marshal(executionOutputs)
+		if err != nil {
+			return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+		}
+		gzWriter.Write(marshalledExecutionOutputs)
+		if err := gzWriter.Close(); err != nil {
+			return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+		}
+		oneTimeApplyOutput = gzOutput.Bytes()
+
+	} else {
+		// For posterity, if a one-time apply was not requested, the one-time apply success is technically false.
+		oneTimeApplyOutput = existingOneTimeOutput
+		oneTimeApplySucceeded = false
+	}
+
+	// ok, let's run the periodic instructions
+
+	periodicOutputs := map[string]PeriodicInstructionOutput{}
+	if len(existingPeriodicOutput) > 0 {
+		buffer := bytes.NewBuffer(existingPeriodicOutput)
+		gzReader, err := gzip.NewReader(buffer)
+		if err != nil {
+			return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+		}
+
+		var objectBuffer bytes.Buffer
+		_, err = io.Copy(&objectBuffer, gzReader)
+		if err != nil {
+			return false, existingOneTimeOutput, false, existingPeriodicOutput, err
+		}
+
+		if err := json.Unmarshal(objectBuffer.Bytes(), &periodicOutputs); err != nil {
+			return false, existingOneTimeOutput, false, existingPeriodicOutput, err
 		}
```

Comment:
This code is very similar to the code above for one-time instructions. Can these be consolidated into a function?

Reply:
Yes, it has been consolidated.